docs/oauth.md

```diff
@@ -0,0 +1,218 @@
+    <!-- Add the scopes needed for OAuth flow
+    E.g. Snowflake need refresh_token scope to issue refresh tokens -->
+    <scopes>scope1</scopes>
+    <scopes>scope2</scopes>
+    <scopes>scope3</scopes>
+    <!-- My Example OAuth Provider supports PKCE, Cutom domain, fixed callback url -->
+    <capabilities>
+        <entry>
+            <key>CAP_PKCE_REQUIRES_CODE_CHALLENGE_METHOD</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_SUPPORTS_CUSTOM_DOMAIN</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_REQUIRE_PKCE</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_FIXED_PORT_IN_CALLBACK_URL</key>
+            <value>true</value>
+        </entry>
+        <entry>
+            <key>CAP_PKCE_REQUIRES_CODE_CHALLENGE_METHOD</key>
+            <value>true</value>
+        </entry>
+    </capabilities>
+    <!-- Map Tableau recognized attr <key> to My Example OAuth Provider response attr <value> -->
+    <accessTokenResponseMaps>
+        <entry>
+            <key>ACCESSTOKEN</key>
+            <value>access_token</value>
+        </entry>
+        <entry>
+            <key>REFRESHTOKEN</key>
+            <value>refresh_token</value>
+        </entry>
+        <entry>
+            <key>access-token-issue-time</key>
+            <value>issued_at</value>
+        </entry>
+        <entry>
+            <key>access-token-expires-in</key>
+            <value>expires_in</value>
+        </entry>
+        <entry>
+            <key>username</key>
+            <value>username</value>
+        </entry>
+    </accessTokenResponseMaps>
+    <!-- No refreshTokenResponseMaps needed since that's same with the accessTokenResponseMaps -->
+ </pluginOAuthConfig>
+
+```
+# OAuth on Tableau Desktop
+
+In Tableau Desktop the connection dialog will look like this.  If a different auth mode is chosen, notice different required fields appear. This example has multiple auth modes supported and one of them is OAuth. 
+
+![Image](../assets/oauth-desktop-dialog.png)
+
+By clicking the sign in button, you will be directed to your OAuth Provider's sign in Url in a default browser on your machine where you can input your credentials, Tableau will get the required attributes back and upon complete you will see this screen:
+
+![Image](../assets/oauth-desktop-complete.png)
+
+# OAuth on Tableau Server
+
+## Configure OAuth Clients on Server
+
+Config your OAuth client on your server: This would be the first step you need to perform for enabling OAuth on Tableau Server, this will setup OAuth Client information to be used on Tableau Server, e.g.:
+```
+tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"snowflake_oauth\", \"oauth.config.client_id\":\"[your_client_id]\", \"oauth.config.client_secret\":\"[your_client_secret]\", \"oauth.config.redirect_uri\":\"[your_redirect_url]\"}]" --force-keys
```

Comment:
I'd make this consistent with the dbClass above.  Also its not clear from this doc that oauth.config.id is the same as the class in the manifest (and dbClass above)